### ISSUE TITLE
QInput label make horizontal scroll appear on iOS 15

### ISSUE THREAD

**Issue by suyundukov** [quasarframework/quasar#10866]
**Describe the bug**
After focusing on QInput vertical scroll appears. The issue is reproducible on iPhone with notch and with the latest iOS version installed.

**Codepen/jsFiddle/Codesandbox (required)**
The issue is reproducible on Quasar docs page.

**To Reproduce**
1. Go to https://quasar.dev/vue-components/input#design
2. Click on "Standard" input (focus)
3. Scroll right or left
4. See vertical scroll

**Expected behavior**
No vertical scroll at all.

**Screenshots**
<img src=https://user-images.githubusercontent.com/14142638/135645007-aa225aef-9b9b-49f0-8de6-c36fdb1384df.png width=250> <img src=https://user-images.githubusercontent.com/14142638/135645012-2646d620-a883-40bd-a732-d38121175673.png width=250>

**Platform (please complete the following information):**
Quasar Version: v1.15.10 (and v2 I guess since it's reproducible on Quasar v2 docs)
@quasar/app Version: v2.2.4

Quasar mode:
  - [x] SPA
  - [x] SSR
  - [ ] PWA
  - [ ] Electron
  - [x] Cordova
  - [x] Capacitor
  - [ ] BEX

Tested on:
  - [x] SPA
  - [x] SSR
  - [ ] PWA
  - [ ] Electron
  - [x] Cordova
  - [x] Capacitor
  - [ ] BEX

OS: iOS
Node: 12.21.0 (but I guess it doesn't matter)
NPM: 6.14.11
Yarn: -
Browsers: Safari (tested only on Safari, but I think it's the same on Chrome or Firefox)
iOS: 15.0

**Additional context**
The issue happens only on some inputs, others inputs works as expected. In the attached video, you can see that vertical scroll appears on first, second and forth input (in "Coloring" block).

https://user-images.githubusercontent.com/14142638/135647209-d65b70d0-e03d-42e6-aa1d-c79acaa90ccf.mp4

**Comment by pdanpdan** [comment]
Hi - can you check if you can reproduce it here: https://pdanpdan.github.io/quasar-docs/vue-components/input#design
It looks like it's the ios overscroll, and it usually show when the page gets zoomed on clicking on an input.
I cannot reproduce it in my branch, even if I cannot identify what could have fixed it, so I need a confirmation that it does not happen to try to identify the fix.

**Comment by aloky** [comment]
@pdanpdan fixed on your branch

**Comment by pdanpdan** [comment]
Now the bad news is I cannot identify any change that fixed it :(
I'll keep looking.

**Comment by suyundukov** [comment]
@pdanpdan can confirm it's fixed on your branch.

**Comment by pdanpdan** [comment]
Can you try something in your code (in the css/app.sass file for example)?
Add these lines:

```sass
.q-field
  &__label
    right: unset
    max-width: 100%
    transition: transform .36s cubic-bezier(.4,0,.2,1), max-width .324s cubic-bezier(.4,0,.2,1)

  &--float .q-field__label
    right: unset
    max-width: 133%
    transition: transform .36s cubic-bezier(.4,0,.2,1), max-width .396s cubic-bezier(.4,0,.2,1)
```

**Comment by suyundukov** [comment]
Yep, that fixed the issue with vertical scroll. To be exact, setting only `max-width` property fixed everything:

```sass
.q-field__label
  max-width: 100%
```

Other styles don't have effect on bug.

**Comment by pdanpdan** [comment]
The rest is for animating the label when it transitions from float to non float
And the unset for right which is used in framework style

**Comment by pdanpdan** [comment]
Please change the title of the issue to `QInput label make horizontal scroll appear on iOS 15` :)

**Comment by rstoenescu** [comment]
Fix will be available in Quasar v2.1.2
